pkg/apis/sources/v1alpha1/interfaces.go

```diff
@@ -58,6 +59,31 @@ func IsMultiTenant(src EventSource) bool {
 	return ok && mt.IsMultiTenant()
 }
 
+// serviceAccountProvider is implemented by source types which are able to
+// influence the shape of the ServiceAccount used by their own receive adapter.
+type serviceAccountProvider interface {
+	WantsOwnServiceAccount() bool
+	ServiceAccountOptions() []func(*corev1.ServiceAccount)
+}
```

Comment:
This is the interface.

It has two methods:
- `WantsOwnServiceAccount()` controls whether or not the object needs its own ServiceAccount or not.
- `ServiceAccountOptions()` can be used to mutate that ServiceAccount if needed (e.g. inject a label or annotation)